File: Interest.java
public class Interest {
    public static double monthlyInterest(double balance, double apr, int days) {
        double dailyRate = apr/365;
        double totalInterest = 0;
        double interest;

        for (int i = 0; i < days; i++) {
            interest = balance * dailyRate;
            totalInterest += interest;
            balance += interest;
        }

        return totalInterest;
    }

    public static void main(String args[]) {
        long start = System.currentTimeMillis();
        // 1,000,000,000
        for (long i = 0; i < 1000000000L; i++)
            monthlyInterest(50000, 0.03, 30);
        long end = System.currentTimeMillis();
        System.out.println(String.format("%.3f seconds", (float) (end-start)/1000));
    }
}

File: interest.go
package main

import "runtime"

func Interest(start float64, rate float64, days int) (total float64) {
	balance := start
	daily_rate := rate / 365
	for i := 0; i < days; i++ {
		interest := balance * daily_rate
		total += interest
		balance += interest
	}

	return total
}
func main() {
	for i := 0; i < 10000; i++ {
        go func(){
            for j := 0; j < 10000000; j++ {
                Interest(50000, 0.03, 30)
            }
        }()
    }
    runtime.Gosched()
}
